model/gen-ai/registry.yaml

```diff
@@ -426,6 +430,57 @@ groups:
         type: int
         brief: The number of dimensions the resulting output embeddings should have.
         examples: [512, 1024]
+      - id: gen_ai.retrieval.type
+        stability: development
+        type: string
+        brief: The type of retrieval operation being performed.
+        examples: ["similarity", "hybrid", "keyword"]
+      - id: gen_ai.retrieval.top_k
+        stability: development
+        type: int
+        brief: The maximum number of results requested.
+        examples: [5, 10, 20]
+      - id: gen_ai.retrieval.documents_retrieved
```

Comment:
do we really need it? is having this attribute helpful without knowing document contents? And/or could we record a skeleton instead of document contents that would capture score + id/name of the doc, so count would be derivable anyway?